src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -796,7 +796,7 @@ private void GenerateFastPathFuncForObject(SourceWriter writer, ContextGeneratio
                     if (defaultCheckType != DefaultCheckType.None)
                     {
                         // Use temporary variable to evaluate property value only once
-                        string localVariableName =  $"__value_{propertyGenSpec.NameSpecifiedInSourceCode}";
+                        string localVariableName =  $"__value_{propertyGenSpec.NameSpecifiedInSourceCode.TrimStart('@')}";
```

Comment:
If we were prefixing it the property name with `__value_`, in the case of `@` should suffix it with `__value_` instead of just trimming?

Reply:
We could. I figured we shouldn't change the naming convention for the servicing fix.

Reply:
`something` and `@something` are equivalent identifiers, e.g.

https://sharplab.io/#v2:CYLg1APgAgTAjAWAFBQMwAJboMIG9nqGYYCWAdgC7oCyAFAJTr5JGvrlUUCmAzlQLzoALDADcBNoSgB2TFG590YdHHEsiAX2QagA

I don't believe it is possible for conflicts to exist in code that compiles.